sotodlib/wiregrid/calibrate.py

```diff
@@ -0,0 +1,407 @@
+
+    Return:
+    - aman : AxisManager
+        This includes fit results for all the inpput detectors
+        cfitval : Estimated parameter values by fitting
+        cfitcov : covariance matrix of the estimated parameters
+        cfitresvar : Residual variance
+    """
+    obs_data = []
+    obs_std = []
+    fit_results = []
+    cfitval = []
+    cfitcov = []
+    cfitresvar = []
+    for _i in range(aman.dets.count):
+        obs_data.append(np.array([aman.wg.Q[:, _i], aman.wg.U[:, _i]]))
+        obs_std = np.array([aman.wg.Qerr[:, _i], aman.wg.Uerr[:, _i]])
+        params_init = _get_initial_param_circle(obs_data[-1])
+        fit_results.append(
+            _comp_plane_fit(obs_data[-1], obs_std, _circle_resid, params_init))
+        cfitval.append(
+            np.array([fit_results[-1].beta[0], fit_results[-1].beta[1], np.sqrt(fit_results[-1].beta[2])]))
+        cfitcov.append(fit_results[-1].cov_beta)
+        cfitresvar.append(fit_results[-1].res_var)
+        pass
+    aman.wg.wrap('cfitval', np.array(cfitval), [(0, 'dets')])
+    aman.wg.wrap('cfitcov', np.array(cfitcov), [(0, 'dets')])
+    aman.wg.wrap('cfitresvar', np.array(cfitresvar), [(0, 'dets')])
+    return aman
+
+def wg_get_lfitres(aman):
+    """
+    Get the results by the linear fitting, which is comparing the reference counts of the wire grid
+    and the angle of the respose in Q+iU plane
+
+    Parameters:
+    - aman : AxisManager
+
+    Return:
+    - aman : AxisManager
+        This includes fit results for all the inpput detectors
+        lfitval : Estimated parameter values by fitting
+        lfitcov : covariance matrix of the estimated parameters
+        lfitresvar : Residual variance
+    """
+    lfitval = []
+    lfiterr = []
+    lfitchi2 = []
+    for _i in range(aman.dets.count):
+        _ang_demodQU = np.rad2deg(np.arctan2(aman.wg.U[:,_i] - aman.wg.cfitval[_i][1], aman.wg.Q[:,_i] - aman.wg.cfitval[_i][0]))
+        yerr_temp = np.sqrt(aman.wg.Qerr[:,_i]**2 + aman.wg.Uerr[:,_i]**2) # this is a temporal value, NOT correct
+        c = cost.LeastSquares((aman.wg.enc_ang_deg+90)%180-90, _ang_demodQU%360, yerr_temp, _linear1d)
+        m = Minuit(c, a=2, b=(_ang_demodQU%360)[np.argsort(aman.wg.enc_ang_deg%180)[0]] -2*min(aman.wg.enc_ang_deg%180))
+        m.limits['a'] = (-2.1,2.1)
+        m.limits['b'] = (-180,480)
+        m.migrad()
+        lfitval.append([m.values[0], m.values[1]%360])
+        lfiterr.append([m.errors[0], m.errors[1]])
+        lfitchi2.append(m.fmin.reduced_chi2)
+        pass
+    aman.wg.wrap('lfitval', np.array(lfitval), [(0, 'dets')])
+    aman.wg.wrap('lfiterr', np.array(lfiterr), [(0, 'dets')])
+    aman.wg.wrap('lfitchi2', np.array(lfitchi2), [(0, 'dets')])
+    return aman
+
+def main():
+    print("This is the module for the calibration using the Sparse Wire Grid.\n \
+        Please excute from sotodlib.wiregrid.calibrate import get_det_angle ")
+    pass
+
+if __name__=='__main__':
+    main()
```

Comment:
This is un-necessary in here. (and remove the `#!/usr/bin/python3` from the top).  Things that should be run from command-line should go into `site-pipeline`.